src/args.rs

```diff
@@ -598,6 +600,20 @@ impl<'a> ArgMatches<'a> {
         }
     }
 
+    /// Returns the given pattern with `\r?` inserted before each `$` if the
+    /// --eol-anchor-include-cr flag is set. Otherwise, the pattern is
+    /// returned unchanged.
+    fn cr_anchr_pattern(&self, pat: String) -> String {
+        if self.is_present("eol-anchor-include-cr") {
+            // If `$` occurs at the end (or beginning), an empty
+            // split-element will be created, causing the `join` to
+            // perform the desired insertion at the correct location.
+            pat.split('$').collect::<Vec<&str>>().join(r"\r?$")
```

Comment:
This should be done by parsing the regex and doing a substitution at the HIR level.

Reply:
@BurntSushi That seems sensible. Are you more amenable to the configuration-option design now, then?

Reply:
Not particularly... I just happened to see this particular issue in passing.

Honestly, this just isn't a priority for me now, and it needs thought.

Reply:
I understand it's not a priority for you, but as I mentioned when I opened the original issue, it makes `$` essentially useless on Windows.

I agree that the *right* solution requires thought, and I can understand why you wouldn't want to introduce a half-baked solution just to deprecate it later. On the other hand, my suggested solution is extremely simple to implement and, I expect, to maintain, and personally I think it is reasonable to introduce command-line options that may be deprecated in favor of a better solution with a warning that this is the case.

I suppose that if you're definitely opposed to a partial solution, I can just continue to maintain my fork until there's a better solution ready to ship, but I don't want to publish a competing crate on Cargo or anything like that, so it's not terribly convenient.

Reply:
Okay, I didn't know that. That's helpful; I'll try that out. Thank you.